feat(modelarmor): Add samples for updating templates

harshnasitcrest opened the pull request:
Added Model Armor Snippets:

- Update Template
- Update Template with labels
- Update Template with metadata

Anu4396 reviewed (approved):
Approved as an SME on the Model Armor team.

harshnasitcrest commented:
> Please fix conflicts before review, thanks.

Done

amanda-tarafa commented:
This is failing. Model Armor has been consistently failing for the last couple of weeks on nightly CI. @harshnasitcrest can we focus on fixing the samples/tests that are already merged first, and then go through these PRs one by one? Thanks.

amanda-tarafa commented on modelarmor/api/ModelArmor.Samples/UpdateTemplateWithMetadata.cs:
```diff
@@ -0,0 +1,81 @@
+
+            // For more details about metadata, refer to the following documentation:
+            // https://cloud.google.com/security-command-center/docs/reference/model-armor/rest/v1/projects.locations.templates#templatemetadata
+            Template.Types.TemplateMetadata templateMetadata =
+                new Template.Types.TemplateMetadata();
+
+            templateMetadata.LogTemplateOperations = true;
+            templateMetadata.LogSanitizeOperations = true;
```
This is more idiomatic, here and everywhere you have similar code.
```suggestion
            var templateMetadata = new Template.Types.TemplateMetadata
            {
                LogTemplateOperations = true;
                LogSanitizeOperations = true;
            }
```

amanda-tarafa commented on modelarmor/api/ModelArmor.Samples.Tests/UpdateTemplateWithMetadataTest.cs:
```diff
@@ -0,0 +1,64 @@
+                templateId: templateName.TemplateId
+            );
+
+            string templateId = TemplateName.Parse(originalTemplate.Name).TemplateId;
```
Here and everywhere:
```suggestion
            string templateId = templateName.TemplateId;
```

amanda-tarafa commented on modelarmor/api/ModelArmor.Samples.Tests/UpdateTemplateTest.cs:
```diff
@@ -0,0 +1,102 @@
+                    && f.ConfidenceLevel == DetectionConfidenceLevel.MediumAndAbove
+            );
+
+            Assert.Contains(raiFilters, f => f.FilterType == RaiFilterType.SexuallyExplicit);
+            Assert.Contains(
```
These are redundant. If it contains the more specific, then it also contains the least specific. Here and in all the asserts above, keep the most specific only.

harshnasitcrest commented:
@amanda-tarafa Thanks for the review. I've addressed all the comments.

Screenshot of successful tests execution on my local machine:
<img width="1172" height="157" alt="Screenshot 2025-08-05 at 1 45 11 PM" src="https://github.com/user-attachments/assets/303e8838-bf93-47be-a65d-f5fd4ea85df1" />
